### PR TITLE
api: directory.withNewFile: make `contents` mandatory

### LINKED ISSUE #3884: API: Creating a new file is annoying
Because of the optional contents, creating a new file looks something like this:

```go
	newDir := c.Directory().WithNewDirectory("/foo").WithNewFile("some-file", dagger.DirectoryWithNewFileOpts{
		Contents: "some-content",
	})
```

By making contents required it would look like this:

```go
	newDir := c.Directory().WithNewDirectory("/foo").WithNewFile("some-file", "some-content")
```

Creating an empty file would still be possible, although a bit quirky:

```go
	newDir := c.Directory().WithNewDirectory("/foo").WithNewFile("some-file", "")
```

From my own dogfooding experience, creating an empty file is extremely rare (never done it) whereas every time I reached for `WithNewFile` it was to inject some contents.

/cc @shykes @vito

### PR DESCRIPTION
**Pull request by aluzzardi**
Fixes #3884

/cc @vikram-dagger @kpenfound @helderco @dolanor

### CODE REVIEW

**Comment by aluzzardi** [comment]
Flagged as `breaking change`

File: sdk/python/src/dagger/api/gen_sync.py [hunk 1]
```diff
@@ -569,11 +569,11 @@ def with_new_directory(self, path: str) -> "Directory":
         _ctx = self._select("withNewDirectory", _args)
         return Directory(_ctx)
 
-    def with_new_file(self, path: str, contents: str | None = None) -> "Directory":
```
Comment (aluzzardi) [inline sdk/python/src/dagger/api/gen_sync.py:572]: /cc @helderco I think `gen_sync.py` is missing from `.gitattributes` :)

Comment (helderco) [reply sdk/python/src/dagger/api/gen_sync.py:572]: Indeed 😊

Comment (helderco) [reply sdk/python/src/dagger/api/gen_sync.py:572]: @aluzzardi Quick update: https://github.com/dagger/dagger/pull/3926

**Review by helderco — APPROVED** [review]
Makes so much sense! :shipit:

**Review by sipsma — APPROVED** [review]

**Pull request merged**